Applications/HexEditor/HexEditor.cpp

```diff
@@ -518,7 +516,7 @@ void HexEditor::paint_event(GUI::PaintEvent& event)
         };
 
         bool is_current_line = (m_position / bytes_per_row()) == i;
-        auto line = String::format("0x%08X", i * bytes_per_row());
+        auto line = String::formatted("{:#010X}", i * bytes_per_row());
```

Comment:
I don't know how `String::formatted` work, but `0x%08X` and `%8X` can't both be equivalent to `{:#010X}`, you must have made a mistake in at least one of these lines.

Reply:
The `#` means "alternative form", for integers this means that prefixes `0x`, `0` or `0b` are added. (If the uppercase `X` is used then `0x` becomes `0X` and for `B` similarly.)

This prefix counts towards the width, so if we want to have 8 significant digits, then we need a width of 10 if we use the alternative form.

I replaced `8X` with `#010X` because this makes it more explicit that the offset is hexadecimal (this only matters for small numbers anyways.)

Reply:
> I replaced 8X with #010X because this makes it more explicit that the offset is hexadecimal (this only matters for small numbers anyways.)

You could have wrote something about it in the commit message or, at least, in the PR message :/

Reply:
Added this to the commit message:

~~~
I changed the formatting for some numbers when it made sense, because I
believe '0x00000020' to be more expressive than '      20' for example.
~~~